### PR TITLE
Primum Non Nocere: The IAC Rescue Ship

### PR DESCRIPTION
**Pull request by LoreCoder**
Adds in a ship of the IAC dedicated to providing medical response to areas in distress, plus a small info pamphlet for it to distribute.

### CODE REVIEW

**Comment by LoreCoder** [comment]
!review

**Review by WickedCybs — CHANGES_REQUESTED** [review]
You need to do a few more things for this.

- Add shuttle navpoints to the fore, aft and starboard sides of the ship.
- Add atmospherics and make sure every room has a vent and scrubber. Windows and airlocks should have emergency shutters mapped in
- Add power. Look at other ships like the headmaster or SSMD patrol vessel to see how that's done if you aren't sure.

File: maps/away/ships/iac/iac_rescue_ship.dm [hunk 1]
```diff
@@ -0,0 +1,106 @@
+/datum/map_template/ruin/away_site/iac_rescue_ship
+	name = "IAC Rescue Ship"
+	description = "The Sanctuary-class rescue ship is a fast response medical vessel, based in large part off of the Asclepius-class medical transport, a much older and more widespread clinic ship, designed to operate mainly between planets rather than in open space. Most Sanctuary-class hulls are heavily refitted to accomodate for the new conditions in the Wildlands, sporting additional thrusters and a hangar bay, created from what was originally a waiting room. However, it is still limited by its origins, having only the bare minimum of crew and atmospherics facilities, as well as being rather obviously unarmed, often needing to return to port for repairs or supplies."
+	suffix = "ships/iac/iac_rescue_ship.dmm"
+	sectors = list(SECTOR_CORP_ZONE, SECTOR_VALLEY_HALE, SECTOR_BADLANDS, SECTOR_NEW_ANKARA, SECTOR_AEMAQ, SECTOR_SRANDMARR, SECTOR_NRRAHRAHUL)
```
Comment (listerla) [inline maps/away/ships/iac/iac_rescue_ship.dm:5]:
```suggestion
	sectors = list(SECTOR_CORP_ZONE, SECTOR_VALLEY_HALE, SECTOR_BADLANDS, SECTOR_SRANDMARR, SECTOR_NRRAHRAHUL)
```
Lore explicitly says they don't operate in Elyra.

**Review by WickedCybs — APPROVED** [review]
Changes look fine.

**Review by Alberyk — APPROVED** [review]

**Review by connorjg1 — CHANGES_REQUESTED** [review]
Tested on a test branch:

The IAC ID's do not appear to work on the IAC vessel doors. You can't open them. Same applies probably to the cabinets and chem storage, but definitely to the Nanomed.

Shuttle control console doesn't work. 

Less important notes:

The bedroom could use another light, the TV is covered in darkness.

Coordinators airlock is unnamed.

Shouldn't the money go in a coordinator safe or cupboard? Having donation money laying with the pamphlets in the open seems like poor money-handling. 

Only one advanced medical kit? If there is serious medical trouble a single one will be used up on two people, could probably use one or two more. 

Giving the coordinator an Advanced Zeng-hu scanner in the office is probably not a bad idea since they're a medical vessel. 

Technician description could probably say they act as either engineering or security. IAC ERT had an armed security role for the medics.

**Comment by LoreCoder** [comment]
> Tested on a test branch:
> 
> Main bugs:
> 
> The IAC ID's do not appear to work on the IAC vessel doors. You can't open them. Not sure if this is something to do with the testing branch hosting? Same applies probably to the cabinets and chem storage, but definitely to the Nanomed.
> 
> Shuttle control console doesn't work.
> 
> Less important notes:
> 
> snip

IAC ID access to everything should be remedied now; minor bugs have all been alleviated as well, +1 AFAK +1 Med voidsuit plus a buff to Coord's gear. As for O- blood, that's stored within the surgery blood locker (which should also be accessible now); right now only issue remaining is the helm/shuttle controls being non-functional. Will fix it as soon as I realize what's up with that.

File: maps/away/ships/iac/iac_rescue_ship.dm [hunk 2]
```diff
@@ -0,0 +1,161 @@
+/datum/map_template/ruin/away_site/iac_rescue_ship
+	name = "IAC Rescue Ship"
+	description = "The Sanctuary-class rescue ship is a fast response medical vessel, based in large part off of the Asclepius-class medical transport, a much older and more widespread clinic ship, designed to operate mainly between planets rather than in open space. Most Sanctuary-class hulls are heavily refitted to accomodate for the new conditions in the Wildlands, sporting additional thrusters and a hangar bay, created from what was originally a waiting room. However, it is still limited by its origins, having only the bare minimum of crew and atmospherics facilities, as well as being rather obviously unarmed, often needing to return to port for repairs or supplies."
+	suffix = "ships/iac/iac_rescue_ship.dmm"
+	sectors = list(SECTOR_CORP_ZONE, SECTOR_VALLEY_HALE, SECTOR_BADLANDS, SECTOR_SRANDMARR, SECTOR_NRRAHRAHUL)
+	spawn_weight = 1
+	spawn_cost = 1
+	id = "iac_rescue_ship"
+	shuttles_to_initialise = list(/datum/shuttle/autodock/overmap/IAC_shuttle)
```
Comment (JohnWildkins) [inline maps/away/ships/iac/iac_rescue_ship.dm:9]:
```suggestion
	shuttles_to_initialise = list(/datum/shuttle/autodock/overmap/iac_shuttle)

```
In general all-lowercase path names are strongly preferred

File: maps/away/ships/iac/iac_rescue_ship.dm [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+	desc = "The Sanctuary-class rescue ship is a fast response medical vessel, based in large part off of the Asclepius-class medical transport, a widespread clinic ship, designed to operate mainly between planets rather than in open space. Most Sanctuary-class hulls are heavily refitted to accomodate for the new conditions in the Wildlands, sporting additional thrusters and a hangar bay, created from what was originally a waiting room. However, it is still limited by its origins, having only the bare minimum of crew and atmospherics facilities, as well as being rather obviously unarmed, often needing to return to port for repairs or supplies."
+	icon_state = "shuttle"
+	moving_state = "shuttle_moving"
+	max_speed = 1/(2 SECONDS)
+	burn_delay = 1 SECONDS
+	vessel_mass = 5000
+	fore_dir = SOUTH
+	vessel_size = SHIP_SIZE_SMALL
+	initial_restricted_waypoints = list(
+		"IAC Ambulance Shuttle" = list("nav_hangar_IAC")
```
Comment (JohnWildkins) [inline maps/away/ships/iac/iac_rescue_ship.dm:74]:
```suggestion
		"IAC Ambulance Shuttle" = list("nav_hangar_iac")

```
ditto

File: maps/away/ships/iac/iac_rescue_ship.dm [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+	)
+
+	initial_generic_waypoints = list(
+		"nav_IAC_rescue_ship_1",
+		"nav_IAC_rescue_ship_2"
```
Comment (JohnWildkins) [inline maps/away/ships/iac/iac_rescue_ship.dm:79]:
```suggestion
	initial_generic_waypoints = list(
		"nav_iac_rescue_ship_1",
		"nav_iac_rescue_ship_2"

```
ditto

File: maps/away/ships/iac/iac_rescue_ship.dm [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+/obj/machinery/computer/shuttle_control/explore/IAC_shuttle
+	name = "shuttle control console"
+	shuttle_tag = "IAC Ambulance Shuttle"
+	req_access = list(access_iac_rescue_ship)
+
+/datum/shuttle/autodock/overmap/IAC_shuttle
+	name = "IAC Ambulance Shuttle"
+	move_time = 20
+	shuttle_area = list(/area/shuttle/iac_shuttle)
+	current_location = "nav_hangar_IAC"
```
Comment (JohnWildkins) [inline maps/away/ships/iac/iac_rescue_ship.dm:143]:
```suggestion
	current_location = "nav_hangar_iac"

```

File: maps/away/ships/iac/iac_rescue_ship.dm [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+	landmark_transition = "nav_transit_IAC_shuttle"
```
Comment (JohnWildkins) [inline maps/away/ships/iac/iac_rescue_ship.dm:144]:
```suggestion
	landmark_transition = "nav_transit_iac_shuttle"

```

File: maps/away/ships/iac/iac_rescue_ship.dm [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+	range = 1
+	fuel_consumption = 2
+	logging_home_tag = "nav_hangar_IAC"
```
Comment (JohnWildkins) [inline maps/away/ships/iac/iac_rescue_ship.dm:147]:
```suggestion
	logging_home_tag = "nav_hangar_iac"

```

File: maps/away/ships/iac/iac_rescue_ship.dm [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+	defer_initialisation = TRUE
+
+/obj/effect/shuttle_landmark/IAC_shuttle/hangar
+	name = "IAC Ambulance Shuttle Hangar"
+	landmark_tag = "nav_hangar_IAC"
```
Comment (JohnWildkins) [inline maps/away/ships/iac/iac_rescue_ship.dm:152]:
```suggestion
	landmark_tag = "nav_hangar_iac"

```

File: maps/away/ships/iac/iac_rescue_ship.dm [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+	docking_controller = "IAC_shuttle_dock"
```
Comment (JohnWildkins) [inline maps/away/ships/iac/iac_rescue_ship.dm:153]:
```suggestion
	docking_controller = "iac_shuttle_dock"

```

File: maps/away/ships/iac/iac_rescue_ship.dm [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+	base_area = /area/ship/iac_rescue_ship
+	base_turf = /turf/simulated/floor/plating
+	movable_flags = MOVABLE_FLAG_EFFECTMOVE
+
+/obj/effect/shuttle_landmark/IAC_shuttle/transit
+	name = "In transit"
+	landmark_tag = "nav_transit_IAC_shuttle"
```
Comment (JohnWildkins) [inline maps/away/ships/iac/iac_rescue_ship.dm:160]:
```suggestion
	landmark_tag = "nav_transit_iac_shuttle"

```

File: maps/sccv_horizon/code/sccv_horizon.dm [hunk 11]
```diff
@@ -117,7 +117,8 @@
 		/datum/shuttle/autodock/overmap/ee_shuttle,
 		/datum/shuttle/autodock/overmap/fsf_shuttle,
 		/datum/shuttle/autodock/overmap/freighter_shuttle,
-		/datum/shuttle/autodock/overmap/kataphract_transport
+		/datum/shuttle/autodock/overmap/kataphract_transport,
+		/datum/shuttle/autodock/overmap/IAC_shuttle
```
Comment (JohnWildkins) [inline maps/sccv_horizon/code/sccv_horizon.dm:121]:
```suggestion
		/datum/shuttle/autodock/overmap/iac_shuttle
```

File: maps/away/ships/iac/iac_rescue_ship.dm [hunk 12]
```diff
@@ -0,0 +1,161 @@
...
+	move_time = 20
+	shuttle_area = list(/area/shuttle/iac_shuttle)
+	current_location = "nav_hangar_IAC"
+	landmark_transition = "nav_transit_IAC_shuttle"
+	range = 1
+	fuel_consumption = 2
+	logging_home_tag = "nav_hangar_IAC"
+	defer_initialisation = TRUE
+
+/obj/effect/shuttle_landmark/IAC_shuttle/hangar
```
Comment (JohnWildkins) [inline maps/away/ships/iac/iac_rescue_ship.dm:150]:
```suggestion
/obj/effect/shuttle_landmark/iac_shuttle/hangar

```

File: maps/away/ships/iac/iac_rescue_ship.dm [hunk 13]
```diff
@@ -0,0 +1,161 @@
...
+	name = "IAC Ambulance Shuttle Hangar"
+	landmark_tag = "nav_hangar_IAC"
+	docking_controller = "IAC_shuttle_dock"
+	base_area = /area/ship/iac_rescue_ship
+	base_turf = /turf/simulated/floor/plating
+	movable_flags = MOVABLE_FLAG_EFFECTMOVE
+
+/obj/effect/shuttle_landmark/IAC_shuttle/transit
```
Comment (JohnWildkins) [inline maps/away/ships/iac/iac_rescue_ship.dm:158]:
```suggestion
/obj/effect/shuttle_landmark/iac_shuttle/transit

```

File: maps/away/ships/iac/iac_rescue_ship.dm [hunk 14]
```diff
@@ -0,0 +1,161 @@
...
+	vessel_mass = 3000 //very inefficient pod
+	fore_dir = NORTH
+	vessel_size = SHIP_SIZE_TINY
+
+/obj/machinery/computer/shuttle_control/explore/IAC_shuttle
+	name = "shuttle control console"
+	shuttle_tag = "IAC Ambulance Shuttle"
+	req_access = list(access_iac_rescue_ship)
+
+/datum/shuttle/autodock/overmap/IAC_shuttle
```
Comment (JohnWildkins) [inline maps/away/ships/iac/iac_rescue_ship.dm:139]:
```suggestion
/datum/shuttle/autodock/overmap/iac_shuttle

```

File: maps/away/ships/iac/iac_rescue_ship.dm [hunk 15]
```diff
@@ -0,0 +1,161 @@
...
+	designation = "Heka"
+	desc = "An inefficient design of ultra-light shuttle known as the Wisp-class. Its only redeeming features are the extreme cheapness of the design and the ease of finding replacement parts. Manufactured by Hephaestus. This one's transponder identifies it as belonging to a Interstellar Aid Corps vessel."
+	shuttle = "IAC Ambulance Shuttle"
+	max_speed = 1/(3 SECONDS)
+	burn_delay = 2 SECONDS
+	vessel_mass = 3000 //very inefficient pod
+	fore_dir = NORTH
+	vessel_size = SHIP_SIZE_TINY
+
+/obj/machinery/computer/shuttle_control/explore/IAC_shuttle
```
Comment (JohnWildkins) [inline maps/away/ships/iac/iac_rescue_ship.dm:134]:
```suggestion
/obj/machinery/computer/shuttle_control/explore/iac_shuttle

```

File: maps/away/ships/iac/iac_rescue_ship.dm [hunk 16]
```diff
@@ -0,0 +1,161 @@
...
+	base_turf = /turf/space/dynamic
+	base_area = /area/space
+
+/obj/effect/shuttle_landmark/iac_rescue_ship/transit
+	name = "In transit"
+	landmark_tag = "nav_transit_iac_rescue_ship"
+	base_turf = /turf/space/transit/north
+
+//shuttle stuff
+/obj/effect/overmap/visitable/ship/landable/IAC_shuttle
```
Comment (JohnWildkins) [inline maps/away/ships/iac/iac_rescue_ship.dm:122]:
```suggestion
/obj/effect/overmap/visitable/ship/landable/iac_shuttle

```

**Review by JohnWildkins — CHANGES_REQUESTED** [review]

File: maps/away/ships/iac/iac_rescue_ship.dm [hunk 17]
```diff
@@ -0,0 +1,161 @@
...
+	name = "IAC Ambulance Shuttle"
+	icon_state = "shuttle2"
+
+//ship stuff
+
+/obj/effect/overmap/visitable/ship/iac_rescue_ship
+	name = "IAC Rescue Ship"
+	class = "IAV"
+	desc = "The Sanctuary-class rescue ship is a fast response medical vessel, based in large part off of the Asclepius-class medical transport, a widespread clinic ship, designed to operate mainly between planets rather than in open space. Most Sanctuary-class hulls are heavily refitted to accomodate for the new conditions in the Wildlands, sporting additional thrusters and a hangar bay, created from what was originally a waiting room. However, it is still limited by its origins, having only the bare minimum of crew and atmospherics facilities, as well as being rather obviously unarmed, often needing to return to port for repairs or supplies."
+	icon_state = "shuttle"
```
Comment (WickedCybs) [inline maps/away/ships/iac/iac_rescue_ship.dm:66]: Don't use the shuttle icon state for the main ship. Use ship.

File: maps/away/ships/iac/iac_rescue_ship.dm [hunk 18]
```diff
@@ -0,0 +1,161 @@
...
+	moving_state = "shuttle_moving"
```
Comment (WickedCybs) [inline maps/away/ships/iac/iac_rescue_ship.dm:67]: Don't use the shuttle icon state for the main ship. Use ship.

File: maps/away/ships/iac/iac_rescue_ship.dm [hunk 19]
```diff
@@ -0,0 +1,178 @@
...
+	icon_state = "ship"
+	moving_state = "ship_moving"
+	max_speed = 1/(2 SECONDS)
+	burn_delay = 1 SECONDS
+	vessel_mass = 5000
+	fore_dir = SOUTH
+	vessel_size = SHIP_SIZE_SMALL
+	initial_restricted_waypoints = list(
+		"IAC Ambulance Shuttle" = list("nav_hangar_iac")
+
```
Comment (SleepyGemmy) [inline maps/away/ships/iac/iac_rescue_ship.dm:76]:
```suggestion

```

File: maps/away/ships/iac/iac_rescue_ship.dm [hunk 20]
```diff
@@ -0,0 +1,178 @@
...
+	)
+
+	initial_generic_waypoints = list(
+		"nav_iac_rescue_ship_1",
+		"nav_iac_rescue_ship_2",
+		"nav_iac_rescue_ship_3",
+		"nav_iac_rescue_ship_4",
+		"nav_iac_rescue_ship_5"
+
```
Comment (SleepyGemmy) [inline maps/away/ships/iac/iac_rescue_ship.dm:85]:
```suggestion

```

File: maps/away/ships/iac/iac_rescue_ship.dm [hunk 21]
```diff
@@ -0,0 +1,178 @@
...
+	base_area = /area/space
+
+/obj/effect/shuttle_landmark/iac_rescue_ship/transit
+	name = "In transit"
+	landmark_tag = "nav_transit_iac_rescue_ship"
+	base_turf = /turf/space/transit/north
+
+//shuttle stuff
+/obj/effect/overmap/visitable/ship/landable/iac_shuttle
+
```
Comment (SleepyGemmy) [inline maps/away/ships/iac/iac_rescue_ship.dm:129]:
```suggestion

```

File: maps/away/ships/iac/iac_rescue_ship.dm [hunk 22]
```diff
@@ -0,0 +1,178 @@
...
+	desc = "An inefficient design of ultra-light shuttle known as the Wisp-class. Its only redeeming features are the extreme cheapness of the design and the ease of finding replacement parts. Manufactured by Hephaestus. This one's transponder identifies it as belonging to a Interstellar Aid Corps vessel."
+	shuttle = "IAC Ambulance Shuttle"
+	max_speed = 1/(3 SECONDS)
+	burn_delay = 2 SECONDS
+	vessel_mass = 3000 //very inefficient pod
+	fore_dir = NORTH
+	vessel_size = SHIP_SIZE_TINY
+
+/obj/machinery/computer/shuttle_control/explore/iac_shuttle
+
```
Comment (SleepyGemmy) [inline maps/away/ships/iac/iac_rescue_ship.dm:142]:
```suggestion

```

File: maps/away/ships/iac/iac_rescue_ship.dm [hunk 23]
```diff
@@ -0,0 +1,178 @@
...
+	name = "shuttle control console"
+	shuttle_tag = "IAC Ambulance Shuttle"
+	req_access = list(access_iac_rescue_ship)
+
+/datum/shuttle/autodock/overmap/iac_shuttle
+
```
Comment (SleepyGemmy) [inline maps/away/ships/iac/iac_rescue_ship.dm:148]:
```suggestion

```

File: maps/away/ships/iac/iac_rescue_ship.dm [hunk 24]
```diff
@@ -0,0 +1,178 @@
...
+	landmark_transition = "nav_transit_iac_shuttle"
+
+	range = 1
+	fuel_consumption = 2
+	logging_home_tag = "nav_hangar_iac"
+
+	defer_initialisation = TRUE
+
+/obj/effect/shuttle_landmark/iac_shuttle/hangar
+
```
Comment (SleepyGemmy) [inline maps/away/ships/iac/iac_rescue_ship.dm:163]:
```suggestion

```

File: maps/away/ships/iac/iac_rescue_ship.dm [hunk 25]
```diff
@@ -0,0 +1,178 @@
...
+	landmark_tag = "nav_hangar_iac"
+
+	docking_controller = "iac_shuttle_dock"
+
+	base_area = /area/ship/iac_rescue_ship/hangar
+	base_turf = /turf/simulated/floor/plating
+	movable_flags = MOVABLE_FLAG_EFFECTMOVE
+
+/obj/effect/shuttle_landmark/iac_shuttle/transit
+
```
Comment (SleepyGemmy) [inline maps/away/ships/iac/iac_rescue_ship.dm:174]:
```suggestion

```

**Review by SleepyGemmy — CHANGES_REQUESTED** [review]
pharmacy windoor has access 33 instead of 211.
consider setting the ship and shuttle control consoles to require access 211.

**Review by SleepyGemmy — APPROVED** [review]

**Review by JohnWildkins — APPROVED** [review]

**Pull request merged**